diff --git a/src/offat/__main__.py b/src/offat/__main__.py
index f73c524..3fc5428 100644
--- a/src/offat/__main__.py
+++ b/src/offat/__main__.py
@@ -44,8 +44,16 @@ def start():
         help='path or url of openapi/swagger specification file',
         required=True,
     )
+    # parser.add_argument(
+    #     '-v', '--version', action='version', version=f"%(prog)s {get_package_version()}"
+    # )
     parser.add_argument(
-        '-v', '--version', action='version', version=f"%(prog)s {get_package_version()}"
+        '--only-get-requests',
+        dest='only_get_requests',
+        help="send only get requests while scanning",
+        action='store_true',
+        default=False,
+        required=False
     )
     parser.add_argument(
         '-rl',
@@ -168,6 +176,7 @@ def start():
         proxies=args.proxies_list,
         capture_failed=args.capture_failed,
         ssl_verify=args.ssl_verify,
+        only_get_requests=args.only_get_requests,
     )
 
 
diff --git a/src/offat/tester/handler.py b/src/offat/tester/handler.py
index 44bf376..e8663bb 100644
--- a/src/offat/tester/handler.py
+++ b/src/offat/tester/handler.py
@@ -28,6 +28,7 @@ def generate_and_run_tests(
     capture_failed: bool = False,
     remove_unused_data: bool = True,
     ssl_verify: bool = True,
+    only_get_requests: bool = False,
 ):
     """
     Generates and runs tests for the provided OAS/Swagger file.
@@ -70,6 +71,7 @@ def generate_and_run_tests(
         headers=req_headers,
         proxies=proxies,
         ssl_verify=ssl_verify,
+        only_get_requests=only_get_requests,
     )
 
     results: list = []
diff --git a/src/offat/tester/runner.py b/src/offat/tester/runner.py
index d61fbdb..80868ce 100644
--- a/src/offat/tester/runner.py
+++ b/src/offat/tester/runner.py
@@ -26,6 +26,7 @@ class TestRunner:
         headers: dict | None = None,
         proxies: list[str] | None = None,
         ssl_verify: bool = True,
+        only_get_requests: bool = False,
     ) -> None:
         self._client = AsyncRequests(
             rate_limit=rate_limit,
@@ -33,6 +34,7 @@ class TestRunner:
             proxies=proxies,
             ssl_verify=ssl_verify,
         )
+        self.only_get_requests = only_get_requests
         self.progress = Progress(console=console)
         self.progress_task_id: TaskID | None = None
 
@@ -125,6 +127,15 @@ class TestRunner:
             )
 
         test_result = test_task
+        if str(http_method).upper() != "GET" and self.only_get_requests:
+            test_result['request_headers'] = []
+            test_result['response_headers'] = []
+            test_result['response_body'] = f"{str(http_method).upper()} request was not sent"
+            test_result['response_status_code'] = -1
+            test_result['redirection'] = ''
+            test_result['error'] = False
+
+            return test_result
         try:
             response = await self._client.request(
                 url=url, method=http_method, *args, **kwargs
